app/controllers/application_controller.rb

```diff
@@ -1,2 +1,35 @@
 class ApplicationController < ActionController::Base
+	def current_ability
+		if buyer_signed_in?
+			@current_ability ||= ::Ability.new(current_buyer)
+		else
+			@current_ability ||= ::Ability.new(current_artist)
+		end
+	end
+
+	rescue_from CanCan::AccessDenied do |exception|
+		puts "WHY"
+		redirect_to '/buyers/sign_in'
```

Comment:
Rn this denies access to viewing artists/works (i believe). Instead, we should make this so that you don't have to be logged in to have view access to an artist's profile or the artist index page, or any works page.

Reply:
Shit so I'm not sure if thats possible can artist/works be works/artist instead? I can look into this though.